src/nvim/os/keyboard.c

```diff
@@ -0,0 +1,93 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+#include "nvim/lua/executor.h"
+#include "nvim/vim.h"
+#include "nvim/version.h"
+
+#ifdef WIN32
+#include <Winuser.h>
+#endif
+
+#ifdef __APPLE__
+#define Boolean Boolean_I_Dont_Care
```

Comment:
OK, sure, I try it too on simple .c file too

Reply:
I think this will not work, because, there was a use of `typedef`, and `typedef` and `#define` are different things
I explain why my code works here, when `nvim/api/private/helpers.h` was included, there was a `typedef bool Boolean;` then when I was going to include `IOKit/*` in that there was a re`typedef` of Boolean, but with other name, so what I did, I changed the name of Boolean to Boolean_I_DONTCARE, what it did, it the places where whas Boolean written, in the code of `IOKit/*` it was replaced by  `Boolean_I_DONTCARE` so, in all places in IOKit, this `Boolean_I_DONTCARE`was used and original Boolean of our neovim `typedef` remain untouched. So It works

Reply:
My mistake, as jamessan said we dealt with this in lang.c

Reply:
Yes, No problem, thanks for your time to check it :smiley: